tests/integration-tests/tests/common/storage/efs_utils.py

```diff
@@ -0,0 +1,67 @@
+import logging
+
+import boto3
+from botocore.exceptions import ClientError
+from retrying import retry
+from time_utils import seconds
+
+from tests.common.networking.security_groups import delete_security_group
+
+
+@retry(stop_max_attempt_number=3, wait_fixed=seconds(5))
+def describe_efs_filesystem(region: str, file_system_id: str):
+    logging.info(f"Describing EFS File System {file_system_id}")
+    try:
+        return _efs(region).describe_file_systems(FileSystemId=file_system_id)["FileSystems"][0]
+    except Exception as e:
+        if isinstance(e, ClientError) and "FileSystemNotFound" in str(e):
+            return None
+        else:
+            logging.error(f"Cannot describe EFS File System {file_system_id}: {e}")
+            raise e
+
+
+@retry(stop_max_attempt_number=10, wait_fixed=seconds(30))
+def delete_efs_filesystem(region: str, file_system_id: str, delete_dependent_resources: bool = True):
+    logging.info(f"Deleting EFS File System {file_system_id}")
+    try:
+        if delete_dependent_resources:
+            mount_targets = describe_efs_mount_targets(region, file_system_id)
+            security_group_ids = set()
+            for mount_target in mount_targets:
+                mount_target_id = mount_target["MountTargetId"]
+                security_groups = describe_mount_target_security_groups(region, mount_target_id)
+                for security_group_id in security_groups:
+                    security_group_ids.add(security_group_id)
+                delete_efs_mount_target(region, mount_target_id)
+            for security_group_id in security_group_ids:
+                delete_security_group(region, security_group_id)
```

Comment:
How does ParallelCluster manage security group of retained storage?

Reply:
When an EFS file system is created by ParallelCluster, the following security group are created along with it:
1. a SG to allow in/outbound traffic within the file system SG itself
2. a SG to allow in/out bound traffic with the head node
3. a SG to allow in/out bound traffic with the compute nodes

When the file system is detached from the cluster with DeletionPolicy=Retain, only the SecurityGroup (1) is retained with the file system. It is like this because that one is the only security group that is strictly related to the file system regardless the specific cluster. The other twos are specific to the cluster so it makes sense to delete them when the file system is detached.

Reply:
Your comment made me think that the FSx file system too should have the equivalent security group to be removed. Doing that